spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/cassandra/CassandraDataAutoConfiguration.java

```diff
@@ -80,10 +113,41 @@ public CassandraSessionFactoryBean session(CassandraConverter converter)
 		session.setCluster(this.cluster);
 		session.setConverter(converter);
 		session.setKeyspaceName(this.properties.getKeyspaceName());
-		session.setSchemaAction(SchemaAction.NONE);
+		session.setSchemaAction(SchemaAction.valueOf(this.properties.getSchemaAction()));
```

Comment:
Our users are expecting an enum (with the relaxed  binding that goes with it) and it's not. We don't use raw strings for such a use case. This has already been raised in the related issue/pr.